### PR TITLE
[DNM] cramjam support

### PR DESCRIPTION
**Pull request by crusaderky**
Blocked by
- cramjam 2.7.0 final release
- A/B tests

### CODE REVIEW

File: distributed/protocol/compression.py [hunk 1]
```diff
@@ -77,10 +80,27 @@ class Compression(NamedTuple):
...
+        None,
     )
     compressions["auto"] = compressions["lz4"]
 
 
+with suppress(ImportError):
+    import cramjam
+
+    # TODO change to 2.7.0. This is a hack to make it work with 2.7.0-rc3.
+    if parse_version(cramjam.__version__) < parse_version("2.6.99"):
```
Comment (crusaderky) [inline distributed/protocol/compression.py:92]: DNM

**Comment by crusaderky** [comment]
The whole `decompress_into` hack would become redundant if we could get rid of sharding.
XREFS:
- #7433 
- #7824

**Comment by crusaderky** [comment]
The A/B tests show a >2x *slowdown* of cramjam.lz4 vs. plain lz4. Did not investigate yet.
https://github.com/coiled/benchmarks/actions/runs/4938359937/jobs/8828024130

**Comment by milesgranger** [comment]
> The A/B tests show a >2x slowdown of cramjam.lz4 vs. plain lz4.

oi, got me nervous. Ran benchmarks again on cramjam's end and all seems normal, as well as your example comparison and got:
```python
=== lz4 ===
0.5676321685314178
75.5 ms ± 574 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
44.8 ms ± 509 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
=== cramjam.lz4 ===
0.5676321685314178
64.3 ms ± 156 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
24.8 ms ± 309 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
9.74 ms ± 22.1 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Nothing immediately stands out to me here why it'd be slower though, but weird edge cases happen. :thinking:

**Comment by crusaderky** [comment]
It's more likely that one test hanged
